.github/workflows/retest.yaml

```
@@ -0,0 +1,26 @@
+name: Detect and Trigger Retest
+on:
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  retest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        if: github.event.comment.author_association == 'MEMBER' && contains(github.event.comment.body, '/retest')
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b #v4.1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Retest
+        if: github.event.comment.author_association == 'MEMBER' && contains(github.event.comment.body, '/retest')
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: ${{ github.repository }}
+          COMMENT: ${{ github.event.comment.body }}
+        run: |
+          gh pr checkout ${{ github.event.issue.number }}
+          SHA_REF=$(git rev-parse HEAD)
+          NAME="${COMMENT#*/retest}"
+          RUNID=$(gh api "repos/$REPO/actions/runs?event=pull_request&head_sha=$SHA_REF" | jq -r ".workflow_runs[] | select(.name == \"$NAME\") | .id"| head -n 1)
```

could potentially make this more specific and filter on the workflow name initially: https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow

```suggestion
          RUN_ID=$(gh api "repos/$REPO/actions/workflows/$WORKFLOW_NAME/runs?per_page=1&event=pull_request&head_sha=$SHA_REF" | jq -r ".workflow_runs[] | .id"| head -n 1)
```

also added `per_page` so that there's only 1 run returned